From f766935b0a70c7f2d243f8e653b9fecc2d337da8 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Mon, 26 Oct 2015 15:47:13 +0100
Subject: [PATCH 1/3] Provide a _SOUP_EXTERN so we ensure the methods get
 externalized

Change the defines to endup using _SOUP_EXTERN so we get
them externalized.

https://bugzilla.gnome.org/show_bug.cgi?id=757146
---
 libsoup/soup-version.h.in | 130 +++++++++++++++++++++++++---------------------
 1 file changed, 70 insertions(+), 60 deletions(-)

diff --git a/libsoup/soup-version.h.in b/libsoup/soup-version.h.in
index 2a48eed..b233540 100644
--- a/libsoup/soup-version.h.in
+++ b/libsoup/soup-version.h.in
@@ -22,6 +22,10 @@ G_BEGIN_DECLS
     (SOUP_MAJOR_VERSION == (major) && SOUP_MINOR_VERSION == (minor) && \
      SOUP_MICRO_VERSION >= (micro)))
 
+#ifndef _SOUP_EXTERN
+#define _SOUP_EXTERN extern
+#endif
+
 guint    soup_get_major_version (void);
 
 guint    soup_get_minor_version (void);
@@ -91,214 +95,220 @@ gboolean soup_check_version     (guint major,
 #error "SOUP_VERSION_MIN_REQUIRED must be >= SOUP_VERSION_2_24"
 #endif
 
+#define SOUP_DEPRECATED G_DEPRECATED
+#define SOUP_DEPRECATED_FOR(f) G_DEPRECATED_FOR(f)
+#define SOUP_UNAVAILABLE(maj,min) G_UNAVAILABLE(maj,min) _SOUP_EXTERN
+
+#define SOUP_AVAILABLE_IN_2_4                   _SOUP_EXTERN
+
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_24
-# define SOUP_DEPRECATED_IN_2_24                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_24_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_24                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_24_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_24
 # define SOUP_DEPRECATED_IN_2_24_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_24
-# define SOUP_AVAILABLE_IN_2_24                 GLIB_UNAVAILABLE(2, 24)
+# define SOUP_AVAILABLE_IN_2_24                 SOUP_UNAVAILABLE(2, 24)
 #else
-# define SOUP_AVAILABLE_IN_2_24
+# define SOUP_AVAILABLE_IN_2_24                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_26
-# define SOUP_DEPRECATED_IN_2_26                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_26_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_26                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_26_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_26
 # define SOUP_DEPRECATED_IN_2_26_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_26
-# define SOUP_AVAILABLE_IN_2_26                 GLIB_UNAVAILABLE(2, 26)
+# define SOUP_AVAILABLE_IN_2_26                 SOUP_UNAVAILABLE(2, 26)
 #else
-# define SOUP_AVAILABLE_IN_2_26
+# define SOUP_AVAILABLE_IN_2_26                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_28
-# define SOUP_DEPRECATED_IN_2_28                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_28_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_28                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_28_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_28
 # define SOUP_DEPRECATED_IN_2_28_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_28
-# define SOUP_AVAILABLE_IN_2_28                 GLIB_UNAVAILABLE(2, 28)
+# define SOUP_AVAILABLE_IN_2_28                 SOUP_UNAVAILABLE(2, 28)
 #else
-# define SOUP_AVAILABLE_IN_2_28
+# define SOUP_AVAILABLE_IN_2_28                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_30
-# define SOUP_DEPRECATED_IN_2_30                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_30_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_30                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_30_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_30
 # define SOUP_DEPRECATED_IN_2_30_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_30
-# define SOUP_AVAILABLE_IN_2_30                 GLIB_UNAVAILABLE(2, 30)
+# define SOUP_AVAILABLE_IN_2_30                 SOUP_UNAVAILABLE(2, 30)
 #else
-# define SOUP_AVAILABLE_IN_2_30
+# define SOUP_AVAILABLE_IN_2_30                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_32
-# define SOUP_DEPRECATED_IN_2_32                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_32_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_32                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_32_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_32
 # define SOUP_DEPRECATED_IN_2_32_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_32
-# define SOUP_AVAILABLE_IN_2_32                 GLIB_UNAVAILABLE(2, 32)
+# define SOUP_AVAILABLE_IN_2_32                 SOUP_UNAVAILABLE(2, 32)
 #else
-# define SOUP_AVAILABLE_IN_2_32
+# define SOUP_AVAILABLE_IN_2_32                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_34
-# define SOUP_DEPRECATED_IN_2_34                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_34_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_34                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_34_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_34
 # define SOUP_DEPRECATED_IN_2_34_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_34
-# define SOUP_AVAILABLE_IN_2_34                 GLIB_UNAVAILABLE(2, 34)
+# define SOUP_AVAILABLE_IN_2_34                 SOUP_UNAVAILABLE(2, 34)
 #else
-# define SOUP_AVAILABLE_IN_2_34
+# define SOUP_AVAILABLE_IN_2_34                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_36
-# define SOUP_DEPRECATED_IN_2_36                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_36_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_36                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_36_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_36
 # define SOUP_DEPRECATED_IN_2_36_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_36
-# define SOUP_AVAILABLE_IN_2_36                 GLIB_UNAVAILABLE(2, 36)
+# define SOUP_AVAILABLE_IN_2_36                 SOUP_UNAVAILABLE(2, 36)
 #else
-# define SOUP_AVAILABLE_IN_2_36
+# define SOUP_AVAILABLE_IN_2_36                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_38
-# define SOUP_DEPRECATED_IN_2_38                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_38_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_38                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_38_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_38
 # define SOUP_DEPRECATED_IN_2_38_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_38
-# define SOUP_AVAILABLE_IN_2_38                 GLIB_UNAVAILABLE(2, 38)
+# define SOUP_AVAILABLE_IN_2_38                 SOUP_UNAVAILABLE(2, 38)
 #else
-# define SOUP_AVAILABLE_IN_2_38
+# define SOUP_AVAILABLE_IN_2_38                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_40
-# define SOUP_DEPRECATED_IN_2_40                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_40_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_40                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_40_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_40
 # define SOUP_DEPRECATED_IN_2_40_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_40
-# define SOUP_AVAILABLE_IN_2_40                 GLIB_UNAVAILABLE(2, 40)
+# define SOUP_AVAILABLE_IN_2_40                 SOUP_UNAVAILABLE(2, 40)
 #else
-# define SOUP_AVAILABLE_IN_2_40
+# define SOUP_AVAILABLE_IN_2_40                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_42
-# define SOUP_DEPRECATED_IN_2_42                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_42_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_42                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_42_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_42
 # define SOUP_DEPRECATED_IN_2_42_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_42
-# define SOUP_AVAILABLE_IN_2_42                 GLIB_UNAVAILABLE(2, 42)
+# define SOUP_AVAILABLE_IN_2_42                 SOUP_UNAVAILABLE(2, 42)
 #else
-# define SOUP_AVAILABLE_IN_2_42
+# define SOUP_AVAILABLE_IN_2_42                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_44
-# define SOUP_DEPRECATED_IN_2_44                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_44_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_44                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_44_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_44
 # define SOUP_DEPRECATED_IN_2_44_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_44
-# define SOUP_AVAILABLE_IN_2_44                 GLIB_UNAVAILABLE(2, 44)
+# define SOUP_AVAILABLE_IN_2_44                 SOUP_UNAVAILABLE(2, 44)
 #else
-# define SOUP_AVAILABLE_IN_2_44
+# define SOUP_AVAILABLE_IN_2_44                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_46
-# define SOUP_DEPRECATED_IN_2_46                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_46_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_46                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_46_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_46
 # define SOUP_DEPRECATED_IN_2_46_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_46
-# define SOUP_AVAILABLE_IN_2_46                 GLIB_UNAVAILABLE(2, 46)
+# define SOUP_AVAILABLE_IN_2_46                 SOUP_UNAVAILABLE(2, 46)
 #else
-# define SOUP_AVAILABLE_IN_2_46
+# define SOUP_AVAILABLE_IN_2_46                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_48
-# define SOUP_DEPRECATED_IN_2_48                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_48_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_48                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_48_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_48
 # define SOUP_DEPRECATED_IN_2_48_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_48
-# define SOUP_AVAILABLE_IN_2_48                 GLIB_UNAVAILABLE(2, 48)
+# define SOUP_AVAILABLE_IN_2_48                 SOUP_UNAVAILABLE(2, 48)
 #else
-# define SOUP_AVAILABLE_IN_2_48
+# define SOUP_AVAILABLE_IN_2_48                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_50
-# define SOUP_DEPRECATED_IN_2_50                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_50_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_50                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_50_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_50
 # define SOUP_DEPRECATED_IN_2_50_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_50
-# define SOUP_AVAILABLE_IN_2_50                 GLIB_UNAVAILABLE(2, 50)
+# define SOUP_AVAILABLE_IN_2_50                 SOUP_UNAVAILABLE(2, 50)
 #else
-# define SOUP_AVAILABLE_IN_2_50
+# define SOUP_AVAILABLE_IN_2_50                 _SOUP_EXTERN
 #endif
 
 #if SOUP_VERSION_MIN_REQUIRED >= SOUP_VERSION_2_52
-# define SOUP_DEPRECATED_IN_2_52                GLIB_DEPRECATED
-# define SOUP_DEPRECATED_IN_2_52_FOR(f)         GLIB_DEPRECATED_FOR(f)
+# define SOUP_DEPRECATED_IN_2_52                SOUP_DEPRECATED
+# define SOUP_DEPRECATED_IN_2_52_FOR(f)         SOUP_DEPRECATED_FOR(f)
 #else
 # define SOUP_DEPRECATED_IN_2_52
 # define SOUP_DEPRECATED_IN_2_52_FOR(f)
 #endif
 
 #if SOUP_VERSION_MAX_ALLOWED < SOUP_VERSION_2_52
-# define SOUP_AVAILABLE_IN_2_52                 GLIB_UNAVAILABLE(2, 52)
+# define SOUP_AVAILABLE_IN_2_52                 SOUP_UNAVAILABLE(2, 52)
 #else
-# define SOUP_AVAILABLE_IN_2_52
+# define SOUP_AVAILABLE_IN_2_52                 _SOUP_EXTERN
 #endif
 
 G_END_DECLS
-- 
2.5.0

